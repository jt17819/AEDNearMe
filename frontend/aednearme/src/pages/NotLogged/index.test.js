import { render, fireEvent, screen, cleanup, within } from '@testing-library/react';
import { MemoryRouter, Router as Router } from 'react-router-dom';
import ReactDOM from 'react-dom';
import { createMemoryHistory } from 'history'
import {default as NotLogged} from './NotLogged';
import axios from 'axios'

const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

describe('Not Logged', () => {

  // beforeEach(() => {
  //   render(<App />, { wrapper: MemoryRouter })
  // });


  // Rendering

  it('renders welcome message', () => {
    render(<NotLogged />, { wrapper: MemoryRouter })
    expect(screen.getByText('Sign in to register new AED')).toBeInTheDocument();
  });

  test('list contains 2 buttons', () => {
    render(<NotLogged />, { wrapper: MemoryRouter });
  
    // const listElement = screen.getAllByRole('button');
    const listItems = screen.getAllByRole('button');
  
    
    expect(listItems.length).toEqual(1);
  });



  // Snapshot

  it('should take a snapshot', () => {
    render(<NotLogged />, { wrapper: MemoryRouter })
    const { asFragment } = render(<NotLogged />, { wrapper: MemoryRouter })
    
    expect(asFragment(<NotLogged />)).toMatchSnapshot()
  })

  

  
})

