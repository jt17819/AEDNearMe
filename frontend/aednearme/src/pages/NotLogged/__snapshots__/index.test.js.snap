// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not Logged should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Sign in to register new AED
    </h1>
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      To register a new AED that is currently not on our map, please sign in/sign up here
    </p>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Login
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
